beacon-chain/core/helpers/beacon_committee.go

```diff
@@ -112,12 +114,16 @@ func BeaconCommitteeFromState(state state.ReadOnlyBeaconState, slot types.Slot,
 //        count=committees_per_slot * SLOTS_PER_EPOCH,
 //    )
 func BeaconCommittee(
+	ctx context.Context,
 	validatorIndices []types.ValidatorIndex,
 	seed [32]byte,
 	slot types.Slot,
 	committeeIndex types.CommitteeIndex,
 ) ([]types.ValidatorIndex, error) {
-	committee, err := committeeCache.Committee(slot, seed, committeeIndex)
+	mLock := async.NewMultilock(fmt.Sprintf("%s-%s", "committee", string(seed[:])))
```

Comment:
One thing that we would benefit from here is either (a) having thorough comments about the choice of keys, or (b) abstracting this into its own small function that explains it. Ideally, this BeaconCommittee function shouldnt deal with the low-level details of how the in-progress cache works using multilock